src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -32,6 +33,24 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+std::string StorageObjectStorage::getPathSample(StorageInMemoryMetadata metadata, ContextPtr context)
+{
+    auto file_iterator = StorageObjectStorageSource::createFileIterator(
```

Comment:
It can throw an exception if path doesn't exists. And it can happen when we have `use_hive_partitioning=1` and try to insert into `s3` for example:
```
insert into function s3('http://localhost:11111/test/hive_partitioning/a=1/b=1/sample.parquet') select * from numbers(10) settings use_hive_partitioning=1

INSERT INTO FUNCTION s3('http://localhost:11111/test/hive_partitioning/a=1/b=1/sample.parquet')
SETTINGS use_hive_partitioning = 1
SELECT *
FROM numbers(10)
SETTINGS use_hive_partitioning = 1

Query id: bae6adad-d6c8-45de-8213-cfa0ce3c3565


Elapsed: 0.005 sec.

Received exception:
Code: 499. DB::Exception: Failed to get object info: No response body.. HTTP response code: 404: while reading hive_partitioning/a=1/b=1/sample.parquet. (S3_ERROR)
``` 

We should override `configuration->getQuerySettings().throw_on_zero_files_match` to `false` and restore initial value later.